.github/workflows/build-redpanda.yml

```diff
@@ -38,11 +39,17 @@ jobs:
           run: ./install-dependencies.sh
         - name: cache directory
           run: mkdir /mnt/ccache
+        - name: get timestamp
+          run: |
+            echo "TIMESTAMP=$(date --utc --iso-8601=seconds)" >> $GITHUB_OUTPUT
+          id: timestamp
         - name: ccache
           uses: actions/cache@v3
           with:
             path: /mnt/ccache
-            key: ${{ github.job }}-${{ matrix.os }}
+            key: ${{ github.job }}-${{ matrix.os }}-${{ steps.timestamp.outputs.TIMESTAMP }}
+            restore-keys: |
+              ${{ github.job }}-${{ matrix.os }}-
```

Comment:
I'm not familiar with this pattern, generally I see something like:

Match current branch, fall back to `dev`. Is there a reason to take the latest build? That could match someone else's random PR.

I might suggest:

```
key: ${{ github.job }}-${{ matrix.os }}-{{ github.ref_name }}-{{ github.sha }}
restore-keys: |
  ${{ github.job }}-${{ matrix.os }}-{{ github.ref_name }}-{{ github.sha }}
  ${{ github.job }}-${{ matrix.os }}-{{ github.ref_name }}-
  ${{ github.job }}-${{ matrix.os }}-dev
```